File: app/src/main/java/com/antyzero/smoksmog/error/ErrorReporter.java
package com.antyzero.smoksmog.error;

import android.support.annotation.StringRes;

/**
 * Common interface for ui elements to report errors
 */
public interface ErrorReporter {

    void report( String message );

    void report( @StringRes int stringId );

    void report( @StringRes int stringId, Object... objects );
}

File: app/src/main/kotlin/com/antyzero/smoksmog/error/ErrorReporter.kt
package com.antyzero.smoksmog.error

import android.support.annotation.StringRes

/**
 * Common interface for ui elements to report errors
 */
interface ErrorReporter {

    fun report(message: String)

    fun report(@StringRes stringId: Int)

    fun report(@StringRes stringId: Int, vararg objects: Any)
}
